storage/src/backend/rocksdb_backend.rs

```diff
@@ -43,101 +29,53 @@ pub struct RocksDBBackendStats {
     cache_total: u64,
 }
 
-impl StorageBackend for RocksDBBackend {
-    fn is_persisted(&self) -> bool {
-        true
-    }
-
-    fn put(&mut self, key: &EntryHash, value: ContextValue) -> Result<bool, StorageBackendError> {
-        let cf = self.inner.cf_handle(&self.column_name).ok_or(
-            StorageBackendError::MissingColumnFamily {
-                name: &self.column_name,
-            },
-        )?;
-
-        self.inner
-            .put_cf_opt(
-                cf,
-                &key.as_ref()[..],
-                &value,
-                &Self::default_write_options(),
-            )
-            .map_err(StorageBackendError::from)
-            .map(|_| true)
+impl GarbageCollector for RocksDBBackend {
```

Comment:
Is it possible to instead of doing this explicitly for each store, have a "NotGarbageCollected" marker trait to which this empty GC logic is implemented for? And stores that don't have GC just impl that trait.

```rust
impl NotGarbageCollected for RocksDBBackend {}
```

Reply:
i ended up using default empty impl;/ what i tried is (https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6ccadea5565853fe30c6ba5a0a038d05) but it doesnt seem to work way i expected it to work

Reply:
@mat13mn I was thinking of something like (using your example):

```rust
impl<T: DefaultFoo> Foo for T {
    fn blah(&self){}
}
```

I don't see why `dyn` is needed.

Reply:
yeah - for some reason compiler wanted it to be `dyn` in my non templated/generic impl (which was bad of course)

```
 --> src/main.rs:7:14
  |
7 | impl Foo for DefaultFoo{
  |              ^^^^^^^^^^ help: use `dyn`: `dyn DefaultFoo`
  |
  = note: `#[warn(bare_trait_objects)]` on by default
```

Reply:
Yeah, because in that case you are trying to implement it for a trait, and not an actual type, in my example I'm implementing it for any type that implements the `DefaultFoo` trait.